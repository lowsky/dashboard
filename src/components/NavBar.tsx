/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { Link } from 'react-router-dom';

export class NavBar extends React.Component {
    state = { burgerActive: false };

    toggleNavbar = () => {
        this.setState(state => ({
            // @ts-ignore
            burgerActive: !state.burgerActive,
        }));
    };

    render() {
        return (
            <nav className="navbar" aria-label="main navigation">
                <div className="navbar-brand">
                    <span className="navbar-item">
                        <a href="https://github.com/lowsky/dashboard/">Dashboard</a>
                    </span>
                    <a
                        role="button"
                        className={'navbar-burger ' + (this.state.burgerActive ? 'is-active' : '')}
                        aria-label="menu"
                        aria-expanded="false"
                        onClick={this.toggleNavbar}>
                        <span aria-hidden="true" />
                        <span aria-hidden="true" />
                        <span aria-hidden="true" />
                    </a>
                </div>
                <div
                    className={'navbar-menu ' + (this.state.burgerActive ? 'is-active' : '')}
                    id="navMenu"
                    aria-label="menu"
                    aria-expanded="false">
                    <div className="navbar-start">
                        <span className="navbar-item">
                            <Link to="/home">Home</Link>
                        </span>
                        <span className="navbar-item">
                            <a href="./story-book/">Storybook</a>
                        </span>
                        <span className="navbar-item">
                            <Link to="/relay">GraphQL + Relay Demo</Link>
                        </span>
                        <span className="navbar-item">
                            <Link to="/restful">RESTful Demo</Link>
                        </span>
                        <span className="navbar-item">
                            <a href="https://github.com/lowsky/dashboard/">
                                <span>
                                    <i className="fab fa-github" />
                                    &nbsp; Github Repo
                                </span>
                            </a>
                        </span>
                    </div>
                </div>
            </nav>
        );
    }
}
