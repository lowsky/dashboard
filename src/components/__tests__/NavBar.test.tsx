import React from 'react';
import ReactDOM from 'react-dom';

import renderer from 'react-test-renderer';

import { NavBar } from '../NavBar';
import { BrowserRouter as Router } from 'react-router-dom';

describe('NavBar component', () => {
    it('should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Router>
                <NavBar />
            </Router>,
            div
        );
    });

    it('should render as expected', () => {
        const props = {
            avatar_url: 'www.avatar.url',
            login: 'login',
            company: 'comp',
        };
        const appContainer = renderer
            .create(
                <Router>
                    <NavBar {...props} />
                </Router>
            )
            .toJSON();
        expect(appContainer).toMatchSnapshot();
    });
});
