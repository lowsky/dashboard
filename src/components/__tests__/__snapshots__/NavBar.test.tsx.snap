// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render as expected 1`] = `
<nav
  aria-label="main navigation"
  className="navbar"
>
  <div
    className="navbar-brand"
  >
    <span
      className="navbar-item"
    >
      <a
        href="https://github.com/lowsky/dashboard/"
      >
        Dashboard
      </a>
    </span>
    <a
      aria-expanded="false"
      aria-label="menu"
      className="navbar-burger "
      onClick={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
  <div
    aria-expanded="false"
    aria-label="menu"
    className="navbar-menu "
    id="navMenu"
  >
    <div
      className="navbar-start"
    >
      <span
        className="navbar-item"
      >
        <a
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
      </span>
      <span
        className="navbar-item"
      >
        <a
          href="./story-book"
        >
          Storybook
        </a>
      </span>
      <span
        className="navbar-item"
      >
        <a
          href="/relay"
          onClick={[Function]}
        >
          GraphQL + Relay Demo
        </a>
      </span>
      <span
        className="navbar-item"
      >
        <a
          href="/restful"
          onClick={[Function]}
        >
          RESTful Demo
        </a>
      </span>
      <span
        className="navbar-item"
      >
        <a
          href="https://github.com/lowsky/dashboard/"
        >
          <span>
            <i
              className="fab fa-github"
            />
            Â  Github Repo
          </span>
        </a>
      </span>
    </div>
  </div>
</nav>
`;
